web/packages/pop-miner/src/types.ts

```diff
@@ -232,6 +232,100 @@ export declare function bitcoinAddressToScriptHash(
   args: BitcoinAddressToScriptHashArgs,
 ): Promise<BitcoinAddressToScriptHashResult>;
 
+/**
+ * Represents a type of event.
+ */
+export type EventType =
+  | 'minerStart'
+  | 'minerStop'
+  | 'mineKeystone'
+  | 'transactionBroadcast';
+
+/**
+ * An event that has been dispatched.
+ */
+export type Event = {
+  type: EventType;
+};
+
+/**
+ * Dispatched when the PoP miner has stopped.
+ */
+export type EventMinerStart = Event & {
+  type: 'minerStart';
+};
+
+/**
+ * Dispatched when the PoP miner has exited.
+ */
+export type EventMinerStop = Event & {
+  type: 'minerStop';
+
+  /**
+   * The error that caused the PoP miner to exit, or null.
+   */
+  error?: Error;
+};
+
+/**
+ * Dispatched when the PoP miner begins mining a keystone.
+ */
+export type EventMineKeystone = Event & {
```

Comment:
Should we call this `EventMineKeystoneStart` based on the description?

Reply:
I think `EventMineKeystone` makes more sense, as it is dispatched with `mineKeystone` is called.
If later we add an event for when a keystone has finished being mined, I think that event would be called `EventMinedKeystone`